<?xml version="1.0"?>
<robot name="planar_3dof_leglike" xmlns:xacro="http://ros.org/wiki/xacro">

  <material name="red">
    <color rgba="0.8 0.0 0.0 1.0"/>
  </material>

  <link name="world" />

  <!-- Base (visualmente igual a link_1) -->
  <link name="base_link">
    <visual>
      <origin rpy="0 0 0" xyz="0 0 0" />
      <geometry>
        <mesh filename="package://lesson_urdf/meshes/visual/arm_link.stl" scale="1 1 1" />
      </geometry>
      <material name="red"/>
    </visual>
  </link>

  <joint name="fixed" type="fixed">
    <parent link="world" />
    <child link="base_link" />
  </joint>

  <!-- Link dummy (no visible) para separar joints sin geometría -->
  <link name="link_twist_mount"/>

  <!-- JOINT 1: DOF tipo pierna (sube/baja): rota TODO lo que sigue alrededor de Y -->
  <joint name="joint_1" type="revolute">
    <parent link="base_link" />
    <child link="link_twist_mount" />
    <!-- SIN offset -->
    <origin xyz="0.0 0.0 0.12" rpy="0 0 0" />
    <!-- si el sentido te queda al revés cambia a: 0 1 0 -->
    <axis xyz="0 1 0" />
    <limit lower="-1.57" upper="1.57" effort="0" velocity="0.5" />
  </joint>

  <!-- Link 1 -->
  <link name="link_1">
    <visual>
      <geometry>
        <mesh filename="package://lesson_urdf/meshes/visual/arm_link.stl" scale="1 1 1" />
      </geometry>
      <material name="red"/>
    </visual>
  </link>

  <!-- JOINT 2: rotación en Z (antes era joint_1) -->
  <joint name="joint_2" type="revolute">
    <parent link="link_twist_mount" />
    <child link="link_1" />
    <origin xyz="0.0 0.0 0.0" rpy="0 0 0" />
    <axis xyz="0 0 1" />
    <limit lower="-1.57" upper="1.57" effort="0" velocity="0.5" />
  </joint>

  <!-- Link 2 -->
  <link name="link_2">
    <visual>
      <geometry>
        <mesh filename="package://lesson_urdf/meshes/visual/arm_link.stl" scale="1 1 1" />
      </geometry>
      <material name="red"/>
    </visual>
  </link>

  <!-- JOINT 3: último joint actuado (antes era joint_2) -->
  <joint name="joint_3" type="revolute">
    <parent link="link_1" />
    <child link="link_2" />
    <!-- al final de link_1 -->
    <origin xyz="0.5 0.0 0.0" rpy="0 0 0" />
    <axis xyz="0 0 1" />
    <limit lower="-1.57" upper="1.57" effort="0" velocity="0.5" />
  </joint>

  <!-- Frame final para cinemática / tf2_echo -->
  <link name="gripper" />
  <joint name="joint_gripper_fixed" type="fixed">
    <parent link="link_2" />
    <child link="gripper" />
    <origin xyz="0.5 0.0 0.0" rpy="0 0 0" />
  </joint>

</robot>
